File: packages/intune/ios/Plugin/Classes/Options/GetPolicyOptions.swift
import Foundation
import Capacitor

@objc public class GetPolicyOptions: NSObject {
    let accountId: String

    init(_ call: CAPPluginCall) throws {
        self.accountId = try GetPolicyOptions.getAccountIdFromCall(call)
    }

    private static func getAccountIdFromCall(_ call: CAPPluginCall) throws -> String {
        guard let accountId = call.getString("accountId") else {
            throw CustomError.accountIdMissing
        }
        return accountId
    }
}

File: packages/intune/android/src/main/java/io/capawesome/capacitorjs/plugins/intune/classes/options/GetPolicyOptions.java
package io.capawesome.capacitorjs.plugins.intune.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.intune.classes.CustomExceptions;

public class GetPolicyOptions {

    @NonNull
    private final String accountId;

    public GetPolicyOptions(@NonNull PluginCall call) throws Exception {
        this.accountId = GetPolicyOptions.getAccountIdFromCall(call);
    }

    @NonNull
    public String getAccountId() {
        return accountId;
    }

    @NonNull
    private static String getAccountIdFromCall(@NonNull PluginCall call) throws Exception {
        String accountId = call.getString("accountId");
        if (accountId == null) {
            throw CustomExceptions.ACCOUNT_ID_MISSING;
        }
        return accountId;
    }
}
